source/opt/constants.cpp

```diff
@@ -185,8 +185,6 @@ Instruction* ConstantManager::BuildInstructionAndAddToModule(
 
 Instruction* ConstantManager::GetDefiningInstruction(
     const Constant* c, uint32_t type_id, Module::inst_iterator* pos) {
-  assert(type_id == 0 ||
-         context()->get_type_mgr()->GetType(type_id) == c->type());
```

Comment:
Won't this start failing if `type_id` is invalid?  Or is that now not possible?

Reply:
If |type_id| is 0, then we just want to use whatever type is found.  There are no invalid values for |type_id|.

Reply:
> If |type_id| is 0, then we just want to use whatever type is found. There are no invalid values for |type_id|.

Ah, quite right.  Thanks.

Reply:
Also if the type for |c| is not the same, then that will be handled by |FindDeclaredConstant|.